apps/hubspot/src/components/FieldModuleNameMapping.tsx

```diff
@@ -42,37 +58,65 @@ const SingleFieldModuleNameMapping = ({
   moduleNameMapping,
   setModuleNameMapping,
   inputDisabled,
+  selectedFields,
+  onValidationChange,
 }: {
   field: SdkField;
   moduleNameMapping: { [fieldId: string]: string };
   setModuleNameMapping: (mapping: { [fieldId: string]: string }) => void;
   inputDisabled: boolean;
+  selectedFields: SdkField[];
+  onValidationChange: (isValid: boolean) => void;
 }) => {
   const [isValid, setIsValid] = useState<boolean>(true);
   const [errorMessage, setErrorMessage] = useState<string>('');
 
-  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const value = e.target.value;
-    setModuleNameMapping({ ...moduleNameMapping, [field.uniqueId]: value });
+  const validateFieldModuleName = (fieldId: string, value: string) => {
+    if (!value) {
+      return { isValid: true, errorMessage: '' };
+    }
 
-    if (MODULE_NAME_PATTERN.test(value)) {
-      setIsValid(true);
-      setErrorMessage('');
-    } else {
-      setIsValid(false);
+    const isFormatValid = MODULE_NAME_PATTERN.test(value);
+
+    if (!isFormatValid) {
       if (value.includes(' ')) {
         const valueWithoutSpaces = value.replace(/\s/g, '');
         if (!MODULE_NAME_PATTERN.test(valueWithoutSpaces)) {
-          setErrorMessage('No spaces or invalid characters');
+          return { isValid: false, errorMessage: 'No spaces or invalid characters' };
         } else {
-          setErrorMessage('No spaces');
+          return { isValid: false, errorMessage: 'No spaces' };
         }
       } else {
-        setErrorMessage('Invalid special character');
+        return { isValid: false, errorMessage: 'Invalid special character' };
```

Comment:
💭 we could use only errorMessage and verify if it has errormessages instead of the isValid